tests/test_errors.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Flask--Resources is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Errors test module."""
+
+import pytest
+from flask import Flask
+
+from flask_resources.resources import CollectionResource, Resource, ResourceConfig
+
+
+# These classes are in the file because they are under test too
+class CustomConfig(ResourceConfig):
+    """Custom resource configuration."""
+
+    item_route = "/custom/<id>"
+    list_route = "/custom/"
+
+
+class CustomResource(CollectionResource):
+    """Custom resource implementation."""
+
+    def __init__(self, *args, **kwargs):
+        """Constructor."""
+        super(CustomResource, self).__init__(config=CustomConfig, *args, **kwargs)
+
+
+@pytest.fixture(scope="module")
+def app():
+    """Application factory fixture."""
+    app_ = Flask(__name__)
+    custom_bp = CustomResource().as_blueprint("custom_resource")
+    app_.register_blueprint(custom_bp)
+    return app_
+
+
+def test_404_if_no_route(client):
+    response = client.get("/foo")
+    assert response.status_code == 404
+
+    response = client.post("/foo")
+    assert response.status_code == 404
+
+    response = client.put("/foo")
+    assert response.status_code == 404
+
+    response = client.patch("/foo")
+    assert response.status_code == 404
+
+    response = client.delete("/foo")
+    assert response.status_code == 404
+
+
+def test_405_if_route_but_no_method(client):
+    # TODO: Only implement MethodView methods for defined Resource methods. Only then
```

Comment:
this is more of a bigger scope discussion. but we (all of us) are setting many todo's in the code, I just fear we will forget. Maybe we can discuss in the standup but something like: before merging all todos/fixmes need to be made into issues?

Reply:
:100: I am guilty on that. I made the issue: https://github.com/inveniosoftware/flask-resources/issues/54 . It's hard in the rush of things, but we definitely should.

Reply:
We are all guilty hehe I just started having a hard tracking my own todos, then imagine if we switch tasks/modules among us will be horrible to follow hehe. We can also do it in July. Mine was more a thought of improvement sort of retrospectivy